pkg/sitewise/client/client.go

```diff
@@ -41,23 +42,139 @@ func NewSitewiseClientForRegion(region string) SitewiseClient {
+			}
+			if len(success) > 0 {
+				for _, successEntry := range output.SuccessEntries {
+					found := false
+					for i, entry := range success {
+						if *entry.EntryId == *successEntry.EntryId {
+							success[i].AssetPropertyValueHistory = append(success[i].AssetPropertyValueHistory, successEntry.AssetPropertyValueHistory...)
+							found = true
+							break
+						}
+					}
+					if !found {
+						success = append(success, successEntry)
+					}
+				}
+			} else {
+				success = append(success, output.SuccessEntries...)
+			}
+			skipped = append(skipped, output.SkippedEntries...)
+			errors = append(errors, output.ErrorEntries...)
+			return true
+		})
+
+	if err != nil {
+		return nil, err
+	}
+
+	return &iotsitewise.BatchGetAssetPropertyValueHistoryOutput{
+		SuccessEntries: success,
+		SkippedEntries: skipped,
+		ErrorEntries:   errors,
+		NextToken:      nil,
+	}, nil
+}
+
 func (c *sitewiseClient) BatchGetAssetPropertyValueHistoryPageAggregation(ctx context.Context, req *iotsitewise.BatchGetAssetPropertyValueHistoryInput, maxPages int, maxResults int) (*iotsitewise.BatchGetAssetPropertyValueHistoryOutput, error) {
 	var (
-		count     = 0
-		numPages  = 0
-		success   []*iotsitewise.BatchGetAssetPropertyValueHistorySuccessEntry
-		skipped   []*iotsitewise.BatchGetAssetPropertyValueHistorySkippedEntry
-		errors    []*iotsitewise.BatchGetAssetPropertyValueHistoryErrorEntry
-		nextToken *string
+		success []*iotsitewise.BatchGetAssetPropertyValueHistorySuccessEntry
+		skipped []*iotsitewise.BatchGetAssetPropertyValueHistorySkippedEntry
+		errors  []*iotsitewise.BatchGetAssetPropertyValueHistoryErrorEntry
 	)
 
-	err := c.BatchGetAssetPropertyValueHistoryPagesWithContext(ctx, req, func(output *iotsitewise.BatchGetAssetPropertyValueHistoryOutput, b bool) bool {
-		numPages++
-		if len(output.SuccessEntries) > 0 {
-			count += len(output.SuccessEntries[0].AssetPropertyValueHistory)
-		}
+	// Divide the time range into 10 chunks and fetch data in parallel
+	startTime := req.Entries[0].StartDate.Unix()
+	endTime := req.Entries[0].EndDate.Unix()
+	intervalCount := 50
+	totalDuration := endTime - startTime
+	intervalDuration := totalDuration / int64(intervalCount)
+
+	// If the time range is less than 10 seconds, we will fetch all data in one request
+	if intervalDuration < 1 {
+		intervalDuration = totalDuration
+		intervalCount = 1
+	}
+
+	var wg sync.WaitGroup
+	results := make([]*iotsitewise.BatchGetAssetPropertyValueHistoryOutput, intervalCount)
+	resultChannel := make(chan IndexedData, intervalCount)
+
+	for i := 0; i < intervalCount; i++ {
+		wg.Add(1)
+		go func(i int) {
+			defer wg.Done()
+			intervalStart := startTime + int64(i)*intervalDuration
+			intervalEnd := intervalStart + intervalDuration
+			if intervalEnd > endTime {
+				intervalEnd = endTime
+			}
+
+			data, err := fetchRawData(ctx, intervalStart, intervalEnd, c, req)
+			if err != nil {
+				fmt.Println(err.Error())
+				return
+			}
```

Comment:
This introduces a significant change in behavior: Currently if any of its paginated requests receives an error, `BatchGetAssetPropertyValueHistoryPageAggregation` returns the error and no data. With this change, all request errors will be ignored and partial data may be returned. I'm not sure how likely this is in practice but it's certainly possible and it could lead to strange behavior.

Handling errors gracefully with parallel requests isn't necessarily simple, but to maintain consistency with the existing behavior, and the behavior of the other methods, it might be better to have the call to `BatchGetAssetPropertyValueHistoryPageAggregation` return no data and an error if any of the parallel calls encounters an error.